streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -312,6 +312,8 @@ public V apply(Change<V> change) {
         Objects.requireNonNull(joiner, "joiner can't be null");
 
         Set<String> allSourceNodes = ensureJoinableWith((AbstractStream<K>) other);
+        enableSendingOldValues();
```

Comment:
I agree that we need to require all the parent processors of the join operator to send old values, originally I thought we were requiring the join processor to also send old values but checking again I think you are doing the right thing: only require parent `KTable` streams to send old values, so this is actually correct.